### PR TITLE
Extend the friction parameters to support viscous and coulomb frictions

### PR DESCRIPTION
**Pull request by sgiraz**
What's new in this PR:

- eomcParser is now able to parse and manage the new friction parameters: `viscousPos`, `viscousNeg`, `coulombPos`, and `coulombNeg` respectively.
- ~~In [`embObjMotionControl.cpp`][2], the `user2raw` functions for the new parameters do not alter the values parsed from the `xml` configuration file (this behavior can be easily changed).~~
- ~~Minor clean up (some trailing spaces have been removed)~~
- add `.vscode` folder to `.gitignore`

**Note:**
- [CanBusMotionControl.cpp][1] hasn't been updated to support the new friction parameters. if is necessary, we can do it.
- The integration has been tested on a custom single joint setup (mc4plus + DC motor + aea2).

[1]: https://github.com/robotology/icub-main/compare/devel...sgiraz:devel?expand=1#diff-6d1c1faf3770675f9c43c2965ac9a91dfcf74ce5788eda2a29cc1ca00e1d5c29
[2]: https://github.com/robotology/icub-main/compare/devel...sgiraz:devel?expand=1#diff-6486113b74ad247f985d0a5ddf9071276a6187ce914437aa7c4adc8eca818ac5R1332-R1335

### CODE REVIEW

File: src/libraries/icubmod/canBusMotionControl/CanBusMotionControl.cpp [hunk 1]
```diff
@@ -1108,6 +1108,10 @@ bool CanBusMotionControlParameters::parsePidsGroup_NewFormat(Bottle& pidsGroup,
     if (!validate(pidsGroup, xtmp, "ko", "Pid ko parameter", _njoints+1))           return false; for (j=0; j<_njoints; j++) myPid[j].offset = xtmp.get(j+1).asFloat64();
     if (!validate(pidsGroup, xtmp, "stictionUp", "Pid stictionUp", _njoints+1))     return false; for (j=0; j<_njoints; j++) myPid[j].stiction_up_val = xtmp.get(j+1).asFloat64();
     if (!validate(pidsGroup, xtmp, "stictionDwn", "Pid stictionDwn", _njoints+1))   return false; for (j=0; j<_njoints; j++) myPid[j].stiction_down_val = xtmp.get(j+1).asFloat64();
+    //if (!validate(pidsGroup, xtmp, "coulombUp", "Pid coulombUp", _njoints+1))   return false;   // TODO: fix
+    //if (!validate(pidsGroup, xtmp, "coulombDwn", "Pid coulombDwn", _njoints+1))   return false; // TODO: fix
+    //if (!validate(pidsGroup, xtmp, "viscousUp", "Pid viscousUp", _njoints+1))   return false;   // TODO: fix
+    //if (!validate(pidsGroup, xtmp, "viscousDwn", "Pid viscousDwn", _njoints+1))   return false; // TODO: fix
```
Comment (pattacini) [inline src/libraries/icubmod/canBusMotionControl/CanBusMotionControl.cpp:1114]: These lines are commented out.
Can we remove them?

**Review by pattacini — COMMENTED** [review]
Hi @sgiraz 
As a general remark, I've seen many modifications that are purely related to formatting. Could you please clean them up?

**Comment by pattacini** [comment]
Hi @sgiraz 

We need to address also the YARP and icub-firmware-shared versions:
- https://github.com/icub-tech-iit/fix/issues/1044#issuecomment-1048632601
- https://github.com/robotology/icub-main/blob/devel/conf/iCubFindDependencies.cmake#L73-L78

**Comment by sgiraz** [comment]
> Hi @sgiraz
> 
> We need to address also the compatibility with the required versions of `yarp` and `icub-firmware-shared`:
> 
> * https://github.com/robotology/icub-main/blob/devel/CMakeLists.txt#L18
> * https://github.com/robotology/icub-main/blob/devel/conf/iCubFindDependencies.cmake#L73-L78

We should probably check with @randaz81, because they could be updated during the official release

**Comment by pattacini** [comment]
>We should probably check with @randaz81, because they could be updated during the official release

`icub-firmware-shared` has been already updated so we can proceed. For YARP, we need to stick to the proper master release, the same.

**Comment by traversaro** [comment]
> > We should probably check with @randaz81, because they could be updated during the official release
> 
> `icub-firmware-shared` has been already updated so we can proceed. For YARP, we need to stick to the proper master release, the same.

Probably we need to keep @Nicogene  in the loop as well given the upcoming release.

**Comment by pattacini** [comment]
>Probably we need to keep @Nicogene in the loop as well given the upcoming distro release.

Good point although we're not forced to include this dev into 2022.02.0. Therefore, any idea on the versioning to avoid conflicts and minimize the friction with the upcoming distro is more than welcome.

**Comment by Nicogene** [comment]
If these changes require YARP 3.6.0 is fine to include them in the upcoming release, otherwise it would be tricky re-build the YARP packages in time

**Pull request merged**